laser_encoders/download_models.py

```diff
@@ -0,0 +1,155 @@
+from language_list import LASER2_LANGUAGE, LASER3_LANGUAGE, SPM_LANGUAGE
+from tqdm import tqdm
+
+assert os.environ.get("LASER"), "Please set the environment variable LASER"
+LASER = os.environ["LASER"]
+sys.path.append(LASER)
+
+from laser_encoders.laser_tokenizer import LaserTokenizer
+from laser_encoders.models import SentenceEncoder
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s",
+)
+logger = logging.getLogger("preprocess")
+
+
+class LaserModelDownloader:
+    def __init__(self, model_dir: str):
+        self.model_dir = model_dir
+        self.url = "https://dl.fbaipublicfiles.com/nllb/laser"
+
+    def download(self, url: str):
+        if os.path.exists(os.path.join(self.model_dir, os.path.basename(url))):
+            logger.info(f" - {os.path.basename(url)} already downloaded")
+        else:
+            logger.info(f" - Downloading {os.path.basename(url)}")
+            response = requests.get(url, stream=True)
+            total_size = int(response.headers.get("Content-Length", 0))
+            progress_bar = tqdm(total=total_size, unit="KB")
+            with open(os.path.join(self.model_dir, os.path.basename(url)), "wb") as f:
+                for chunk in response.iter_content(chunk_size=1024):
+                    f.write(chunk)
+                    progress_bar.update(len(chunk))
+            progress_bar.close()
+
+    def download_laser2(self):
+        self.download(f"{self.url}/laser2.pt")
+        self.download(f"{self.url}/laser2.spm")
+        self.download(f"{self.url}/laser2.cvocab")
+
+    def download_laser3(self, lang: str, version: str = "v1", spm: bool = False):
+        lang_3_4 = LASER3_LANGUAGE[lang]
+        if isinstance(lang_3_4, tuple):
+            options = ", ".join(f"'{opt}'" for opt in lang_3_4)
+            logger.info(
+                f"Language '{lang_3_4}' has multiple options: {options}. Please specify using --lang-model."
+            )
+            return
+
+        self.download(f"{self.url}/laser3-{lang_3_4}.{version}.pt")
+        if spm:
+            if lang_3_4 in SPM_LANGUAGE:
+                self.download(f"{self.url}/laser3-{lang_3_4}.{version}.spm")
+                self.download(f"{self.url}/laser3-{lang_3_4}.{version}.cvocab")
+            else:
+                self.download(f"{self.url}/laser2.spm")
+                self.download(f"{self.url}/laser2.cvocab")
+
+    def main(self, args):
+        if args.lang_model in LASER2_LANGUAGE:
+            self.download_laser2()
+        elif args.lang_model in LASER3_LANGUAGE:
+            self.download_laser3(
+                lang=args.lang_model, version=args.version, spm=args.spm
+            )
+        else:
+            raise ValueError(
+                f"Unsupported language name: {args.lang_model}. Please specify a supported language name using --lang-model."
+            )
+
+
+def download_model(model_dir, version, lang_model, spm):
```

Comment:
Why `lang_model` and not just `lang`?
The prefix `_model` in the variable name looks confusing to me.

Reply:
It's meant to denote the language for a model, hence lang_model but i can change it if it's confusing

Reply:
> It's meant to denote the language for a model,

Are we dealing here with any languages that are NOT for the model?

> i can change it if it's confusing

When I see `lang_model`, I read it not as "language for a model", but as a "[language model](https://en.wikipedia.org/wiki/Language_model)", which is a totally different concept. So yes, it is confusing.